examples/.c3/components/backend/COM-002-db-pool.md

```diff
@@ -0,0 +1,82 @@
+---
+id: COM-002-db-pool
+title: DB Pool (Resource)
+summary: >
+  PostgreSQL connection pooling for the backend. Owns stack/config, retry behavior, and surfaces errors to routes.
+---
+
+# [COM-002-db-pool] DB Pool (Resource)
+
+## Overview {#com-002-overview}
+- Manages pooled PostgreSQL connections for the backend container.
+- Implements CTX SQL protocol from [CON-001-backend#con-001-protocols](../../containers/CON-001-backend.md#con-001-protocols).
+
+## Stack {#com-002-stack}
+- Library: `pg` 8.11.x
+- Why: Native driver, stable, supports pooling and async/await
+
+## Configuration {#com-002-config}
+| Env Var | Dev | Prod | Why |
+|---------|-----|------|-----|
+| DB_URL | postgres://localhost/tasks | ${DB_URL} | Connection target |
+| DB_POOL_MIN | 2 | 10 | Baseline connections |
+| DB_POOL_MAX | 10 | 50 | Handle higher load |
+| DB_IDLE_TIMEOUT | 30s | 10s | Release idle connections |
+| DB_CONNECTION_TIMEOUT | 5s | 10s | Tolerate slower prod networks |
+
+### Config Loading {#com-002-config-loading}
+```typescript
+import { z } from 'zod';
+
+const schema = z.object({
+  url: z.string().url(),
+  min: z.coerce.number().default(2),
+  max: z.coerce.number().default(10),
+  idleTimeoutMillis: z.coerce.number().default(30000),
+  connectionTimeoutMillis: z.coerce.number().default(5000),
+});
+
+export const dbConfig = schema.parse({
+  url: process.env.DB_URL,
+  min: process.env.DB_POOL_MIN,
+  max: process.env.DB_POOL_MAX,
+  idleTimeoutMillis: process.env.DB_IDLE_TIMEOUT,
+  connectionTimeoutMillis: process.env.DB_CONNECTION_TIMEOUT,
+});
+```
+
+## Interfaces & Types {#com-002-interfaces}
```

Comment:
Good addition\! The Interfaces & Types section is something I didn't include in my PR #3. This makes the component contract explicit. Consider adding this to the component template checklist.